cmd/soroban-cli/src/commands/init.rs

```diff
@@ -0,0 +1,329 @@
+    Logging,
+    SimpleAccount,
+    SingleOffer,
+    Timelock,
+    Token,
+    UpgradeableContract,
+    None,
+}
+
+impl ToString for ExampleContract {
+    fn to_string(&self) -> String {
+        match self {
+            ExampleContract::Account => String::from("account"),
+            ExampleContract::Alloc => String::from("alloc"),
+            ExampleContract::AtomicMultiswap => String::from("atomic_multiswap"),
+            ExampleContract::AtomicSwap => String::from("atomic_swap"),
+            ExampleContract::Auth => String::from("auth"),
+            ExampleContract::CrossContract => String::from("cross_contract"),
+            ExampleContract::CustomTypes => String::from("custom_types"),
+            ExampleContract::DeepContractAuth => String::from("deep_contract_auth"),
+            ExampleContract::Deployer => String::from("deployer"),
+            ExampleContract::Errors => String::from("errors"),
+            ExampleContract::Events => String::from("events"),
+            ExampleContract::Fuzzing => String::from("fuzzing"),
+            ExampleContract::HelloWorld => String::from("hello_world"),
+            ExampleContract::Increment => String::from("increment"),
+            ExampleContract::LiquidityPool => String::from("liquidity_pool"),
+            ExampleContract::Logging => String::from("logging"),
+            ExampleContract::SimpleAccount => String::from("simple_account"),
+            ExampleContract::SingleOffer => String::from("single_offer"),
+            ExampleContract::Timelock => String::from("timelock"),
+            ExampleContract::Token => String::from("token"),
+            ExampleContract::UpgradeableContract => String::from("upgradeable_contract"),
+            ExampleContract::None => String::from("none"),
+        }
+    }
+}
+
+#[derive(Parser, Debug, Clone)]
+#[group(skip)]
+pub struct Cmd {
+    pub project_path: String,
+
+    /// optional flag to specify soroban example contracts to include
+    #[arg(short, long, num_args = 1..=20, default_value = "none")]
+    pub with_contract: Vec<ExampleContract>,
+}
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("Io error: {0}")]
+    CreateDirError(#[from] io::Error),
+
+    // the gix::clone::Error is too large to include in the error enum as is, so we wrap it in a Box
+    #[error("Failed to clone the template repository")]
+    CloneError(#[from] Box<gix::clone::Error>),
+
+    // the gix::clone::fetch::Error is too large to include in the error enum as is, so we wrap it in a Box
+    #[error("Failed to fetch the template repository: {0}")]
+    FetchError(#[from] Box<gix::clone::fetch::Error>),
+
+    #[error("Failed to checkout the template repository: {0}")]
+    CheckoutError(#[from] gix::clone::checkout::main_worktree::Error),
+
+    #[error("Failed to parse Cargo.toml: {0}")]
+    TomlParseError(#[from] TomlError),
+}
+
+const TEMPLATE_URL: &str = "https://github.com/AhaLabs/soroban-init-template.git";
```

Comment:
This is fine for now, but may be worth having this be a command line flag in the future to be able to use different templates

Reply:
Great point! I'll plan to wrap a `--template` flag into a follow up PR.